### PR TITLE
feat(FEC-10290): upgrade NPM packages

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

WIP

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -34,6 +30,9 @@ module.exports = {
   },
   devtool: 'source-map',
   plugins: plugins,
+  optimization: {
+    minimize: PROD
```
Comment (OrenMe) [inline webpack.config.js:34]: is this needed? just use the `--mode production` or `--mode development` in package.json scripts

File: src/components/video-player/video-player.js [hunk 2]
```diff
@@ -113,7 +110,7 @@ class VideoPlayer extends Component {
   render(): React$Element<any> {
     const {videoStyles, targetId} = this.props;
 
-    return <div id={`${targetId}-video`} className={style.videoPlayer} style={videoStyles} ref={this._setRef} />;
+    return <div id={`${targetId}-video`} className={style.videoPlayer} style={videoStyles} ref={ref => (ref ? this._setRef(ref) : undefined)} />;
```
Comment (OrenMe) [inline src/components/video-player/video-player.js:113]: better not to use anonymous function for ref, can this be handled inside the setRef
also, why do we have logic in setRef? seems problematic. @yairans FYI
Can we move this to one of the lifecycle methods? if so lets handle it in another ticket.

File: src/components/tooltip/tooltip.js [hunk 3]
```diff
@@ -165,7 +164,7 @@ class Tooltip extends Component {
         <span
           style={{maxWidth: props.maxWidth}}
           ref={el => {
```
Comment (OrenMe) [inline src/components/tooltip/tooltip.js:166]: why use this patten?
Also, see other comment about ref - better use the `createRef` preact utility for this.

File: src/components/share-overlay/share-overlay.js [hunk 4]
```diff
@@ -99,7 +99,7 @@ const ShareUrl = (props: Object): React$Element<any> => {
   return (
     <div className={props.copy ? style.copyUrlRow : ''}>
       <div className={[style.formGroup, style.hasIcon, style.inputCopyUrl].join(' ')} style="width: 350px;">
-        <input tabIndex="-1" type="text" ref={c => (_ref = c)} className={style.formControl} value={props.shareUrl} readOnly />
+        <input tabIndex="-1" type="text" ref={c => (c ? (_ref = c) : undefined)} className={style.formControl} value={props.shareUrl} readOnly />
```
Comment (OrenMe) [inline src/components/share-overlay/share-overlay.js:102]: same here for usage with createRef

File: flow-typed/types/ui-options.js [hunk 5]
```diff
@@ -8,5 +8,8 @@ declare type UIOptionsObject = {
   components?: ComponentsConfig,
   uiComponents?: Array<PKUIComponent>,
   translations?: {[langKey: string]: Object},
-  locale?: string
+  locale?: string,
+  disable?: boolean,
+  css?: string,
```
Comment (OrenMe) [inline flow-typed/types/ui-options.js:13]: these two are actually only part of kaltura player ui-wrapper.
If we would to export types then we would have inherited from them in wrapper, and add these two there

**Comment by Yuvalke** [comment]
Why do we have an event manager and fake event implementation here and in `playkit-js`?

**Comment by Yuvalke** [comment]
Why don't we create a

**Comment by dan-ziv** [comment]
> Why do we have an event manager and fake event implementation here and in `playkit-js`?

**Comment by OrenMe** [comment]
@dan-ziv did you mean to close this?

**Comment by Yuvalke** [comment]
> > Why do we have an event manager and fake event implementation here and in `playkit-js`?
> 
> @Yuvalke because we didn't want that playkit-js will be a dependency of the ui.
> I think we should create playkit-js-utils and import it from there.

@dan-ziv agree

**Pull request closed without merging**